operator/v1/types_console.go

```diff
@@ -375,6 +380,28 @@ const (
 	BrandROSA Brand = "ROSA"
 )
 
+// IngressCustomization allows cluster admin to configure alternative ingress for the console.
+type IngressCustomization struct {
+	// consoleURL is a URL to be used as the base console address.
+	// This field is intended for clusters without ingress capability,
+	// where access to routes is not possible.
+	// Must use the HTTPS scheme.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="isURL(self)",message="console url must be a valid absolute URL"
+	// +kubebuilder:validation:XValidation:rule="url(self).getScheme() == 'https'",message="console url scheme must be https"
```

Comment:
I'm not sure how this would error in the case that it isn't a URL, need to check, do we run the rules in order or not.

A fix to remove ambiguity would be to `isURL(self) && url(self).getScheme() == 'https'`

Your tests indicate they run in order so probably fine as is

Reply:
I wanted more detailed error messages so I decided not to combine them. I added a test case to validate that the order of the validations is respected (when the url is invalid and has no scheme).